shared/public/system.h

```diff
@@ -103,7 +104,7 @@ namespace ink
 #endif
 
 #ifdef INK_ENABLE_STL
-	using ink_exception = std::exception;
+	using ink_exception = std::runtime_error;
```

Comment:
`std::runtime_error` is more precise for this use case and can be constructed from string